src/Deserialization.cpp

```diff
@@ -0,0 +1,1325 @@
+        code = halide_type_handle;
+        break;
+    case TypeCode::TypeCode_BFloat:
+        code = halide_type_bfloat;
+        break;
+    default:
+        user_error << "unknown type code " << code_deserialized << "\n";
+    }
+
+    return Type(code, bits, lanes);
+}
+
+void Deserializer::deserialize_function(const Serialize::Func *function, Function &hl_function) {
+    user_assert(function != nullptr) << "deserializing a null Function\n";
+    const std::string name = deserialize_string(function->name());
+    const std::string origin_name = deserialize_string(function->origin_name());
+    const std::vector<Type> output_types = deserialize_vector<Serialize::Type, Type>(function->output_types(), &Deserializer::deserialize_type);
+    const std::vector<Type> required_types = deserialize_vector<Serialize::Type, Type>(function->required_types(), &Deserializer::deserialize_type);
+    const int required_dim = function->required_dims();
+    const std::vector<std::string> args = deserialize_vector<flatbuffers::String, std::string>(function->args(), &Deserializer::deserialize_string);
+    const auto func_schedule = deserialize_func_schedule(function->func_schedule());
+    const auto init_def = deserialize_definition(function->init_def());
+    const std::vector<Definition> updates = deserialize_vector<Serialize::Definition, Definition>(function->updates(), &Deserializer::deserialize_definition);
+    const std::string debug_file = deserialize_string(function->debug_file());
+    std::vector<Parameter> output_buffers;
+    output_buffers.reserve(function->output_buffers_names()->size());
+    for (const auto &output_buffer_name_serialized : *function->output_buffers_names()) {
+        auto output_buffer_name = deserialize_string(output_buffer_name_serialized);
+        Parameter output_buffer;
+        if (auto it = external_params.find(output_buffer_name); it != external_params.end()) {
+            output_buffer = it->second;
+        } else if (auto it = parameters_in_pipeline.find(output_buffer_name); it != parameters_in_pipeline.end()) {
+            output_buffer = it->second;
+        }
+        output_buffers.push_back(output_buffer);
+    }
+    const std::vector<ExternFuncArgument> extern_arguments = deserialize_vector<Serialize::ExternFuncArgument, ExternFuncArgument>(function->extern_arguments(), &Deserializer::deserialize_extern_func_argument);
+    const std::string extern_function_name = deserialize_string(function->extern_function_name());
+    const auto name_mangling = deserialize_name_mangling(function->extern_mangling());
+    const auto extern_function_device_api = deserialize_device_api(function->extern_function_device_api());
+    const auto extern_proxy_expr = deserialize_expr(function->extern_proxy_expr_type(), function->extern_proxy_expr());
+    const bool trace_loads = function->trace_loads(), trace_stores = function->trace_stores(), trace_realizations = function->trace_realizations();
```

Comment:
In Halide, prefer to define/init each var on a separate line, it's easier to read:

const bool trace_loads = ...
const bool trace_stores = ...

etc